pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -338,7 +341,7 @@ def on_predict_batch_start(
         self.predict_progress_bar.set_description(f"{self.predict_description} DataLoader {dataloader_idx}")
 
     def on_predict_batch_end(self, *_: Any) -> None:
-        if self._should_update(self.predict_batch_idx, self.total_predict_batches_current_dataloader):
+        if self._should_update(self.predict_batch_idx, self.predict_progress_bar.total):
```

Comment:
we assign `total` to the progress bar already, so we can extract it from there itself. Just makes it less error-prone.